### ISSUE TITLE
[web][a11y] issues with screenreaders (Narrator and Ocra)

### ISSUE THREAD

**Issue by nturgut** [flutter/flutter#66383]
Due to the error happening using the [tab key](https://github.com/flutter/flutter/issues/52106) when screen reader is on, the screen readers which uses `tab` as the main navigator key (such as Narrator), a11y does not work (unless the user explicitly changes the modifier keys).

**Comment by nturgut** [comment]
blocker: https://github.com/flutter/flutter/issues/52106

**Comment by nturgut** [comment]
@mariamhas 
I am removing #52106 as blocker for this issue. Looks like we currently able to use CapsLock as the modifier key. So unlike our tests on September, I was able to use a11y with Flutter web.

From the tests I did with narrator the following areas works:
- Enabling a11y in desktop
- Iterating over controllers (buttons, checkboxes)
- Reading a text content
- Interacting with a controller
Areas that seemed to have issues:
- Moving between text fields
- The focus area is not correct. I also created a jsfiddle example, and looks also for regular html+css web pages focus areas are slightly off for Narrator. The gap is bigger for flutter desktop though.

I'll come back to this issue again after fixing the text area related issues.

**Comment by mariamhas** [comment]
SGTM
